core/src/main/java/de/danoeh/antennapod/core/feed/FeedMedia.java

```diff
@@ -467,16 +413,16 @@ public boolean streamAvailable() {
     }
 
     @Override
-    public void saveCurrentPosition(SharedPreferences pref, int newPosition, long timeStamp) {
-        if(item != null && item.isNew()) {
-            DBWriter.markItemPlayed(FeedItem.UNPLAYED, item.getId());
-        }
-        setPosition(newPosition);
-        setLastPlayedTime(timeStamp);
+    public void saveCurrentPosition(SharedPreferences sharedPreferences, PlaybackPosition currentPosition) {
```

Comment:
This looks a bit dangerous. What if some method outside uses this and does not call the DBWriter methods? I think it could be more safe to remove the method.

Reply:
Agree, but we have similar issue here to shownotes. I need to think how to remove this method from `Playable` interface and move logic to `PlayableUtil', if it make sense or move it somewhere else. PlaybackService is calling saveCurrentPossition in few scenarios. We want to save in DB information about playback information  while playback is paused etc, we could create an Observer who will write to DB on certain events. To do that i need more information about business logic of playback.